packages/@aws-cdk/aws-cloudfront/lib/origin.ts

```diff
@@ -139,6 +148,7 @@ export abstract class OriginBase implements IOrigin {
     this.customHeaders = props.customHeaders;
     this.originShieldRegion = props.originShieldRegion;
     this.originId = props.originId;
+    this.originShieldEnabled = props.originShieldEnabled;
```

Comment:
This code is fine in principle. However boolean variables that might be also be `undefined`, can easily turn into bugs when not taken care of in if statements. Consider this code:
```
if (this.originShieldEnabled) {
  // do stuff when enabled
} else {
 // do stuff when disabled
}
```
In this case it seems to be the default is `true`, so the above would be broken when the value is `undefined`.

I'd recommend to always set an explicit default in these cases and adjust the private types accordingly:
```suggestion
    this.originShieldEnabled = props.originShieldEnabled ?? true;
```

(L136, L175 and L213 will need a change)